pkg/build/api/v1/types.go

```diff
@@ -163,6 +163,9 @@ type BuildSource struct {
 	// repository.
 	ContextDir string `json:"contextDir,omitempty" description:"specifies sub-directory where the source code for the application exists, allows for sources to be built from a directory other than the root of a repository"`
 
+	// DockerfileName is the name of the Dockerfile that will be used to build the Docker image.
+	DockerfileName string `json:"dockerfileName,omitempty" description:"specifies the name of the Dockerfile to use for building the Docker image"`
```

Comment:
Even in that case I'd still say it should be part of DBS struct. Some time ago I've proposed bringing up some of the common fields of current specialized structs but that was declined, since it's better to have them at the specialization level. I'm pretty sure the same argument applies here as well. Especially that in future has no time-frame, imho.